test/test_unresolved_links.cpp

```diff
@@ -794,4 +806,69 @@ TEST(Links_ManyObjects)
     tr->commit();
 }
 
+TEST(Unresolved_PerformanceLinks)
```

Comment:
It would be ideal to have a micro benchmark for this so we can track it long term to detect regressions, but that's not blocking for me.